src/Qir/Samples/test-qir-samples.ps1

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+& (Join-Path $PSScriptRoot .. .. .. build set-env.ps1)
+$all_ok = $true
+
+function Test-Sample {
+    [CmdletBinding()]
+    param (
+        [Parameter()]
+        [string]
+        $Sample
+    )
+
+    Write-Host "##[info]Test QIR Samples ($Sample)"
+
+    Push-Location (Join-Path $PSScriptRoot build $Env:BUILD_CONFIGURATION $Sample)
+
+    ctest --verbose
+
+    if ($LastExitCode -ne 0) {
+        Write-Host "##vso[task.logissue type=error;]Failed to test QIR Samples ($Sample)"
+        $script:all_ok = $False
+    }
```

Comment:
(Just curiosity)
I assume that `$script:all_ok` means the same as `$all_ok` (earlier and later in this script and in `build-qir-samples.ps1`)

Reply:
I believe I got the point of the `script:` fragment. It prevents creating another copy of `$all_ok` in a nested scope. If that's the case then please consider this as resolved.

Reply:
PowerShell has some weird scoping rules that can cause problems if you aren't careful. Because this is in a function, using just `$all_ok` would create a local variable and assign the boolean value, not update the variable used elsewhere. The explicit `script` ensures PowerShell knows we intend to refer to the script-wide scope and will correctly update the variable. This is distinct from `global` which will set variables in the whole PowerShell session and could affect the execution of later scripts, and `env` which sets environment variables that will be inherited by launched processes. I've seen missing `script` lead to some very subtle bugs in the past!

See [about_Scoping](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_scopes?view=powershell-7.1) for more info.